### PR TITLE
Issue#88 Create Run Bundle

### PR DESCRIPTION
**Pull request by ChengshuLi**
Still work in progress. 

To-do: 
1. when the user add a new bundle, it should refresh itself
2. when the user switch to a new worksheet, it should refresh itself. 

@percyliang

### CODE REVIEW

**Comment by ChengshuLi** [comment]
@percyliang Feel free to have a look.

File: codalab/apps/web/static/js/worksheet/run_bundle_builder.jsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+      type='default'
+      handleClick={this.toggleBuilder}
+    />
+    return (
+      <div>
+        <div id='run-bundle-builder'>
+          <span className='close' onClick={this.toggleBuilder}>×</span>
+          <div className='pop-up-title'>Create Run Bundle</div>
+          <div className='run-bundle-container'>
+            <div className='run-bundle-text pop-up-text'>Step 1: Specify your dependencies. You can use the file browser below to drill down.</div>
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/run_bundle_builder.jsx]: Rephrase: "Select any dependencies (either bundles or the files/directories inside)."

File: codalab/apps/web/static/js/worksheet/run_bundle_builder.jsx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+            <div id='bundle-browser'>
+              {bundles_html}
+            </div>
+          </div>
+          <div className='run-bundle-container'>
+            <div className='run-bundle-text pop-up-text'>Step 2: Run your command. Your dependencies have been put into the same environment for you. </div>
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/run_bundle_builder.jsx]: Rephrase: "Enter a shell command, which will be run in a directory with the dependencies (rename if desired)."

File: codalab/apps/web/static/js/worksheet/run_bundle_builder.jsx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+              </tr>
+              );
+          }
+        }.bind(this));
+      }
+    }.bind(this));
+
+    if (rows.length === 0) {
+      return (<div className='pop-up-text'>
+        You don't have any bundle in this worksheet
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/run_bundle_builder.jsx]: "You don't have any bundles in this worksheet."

File: codalab/apps/web/static/js/worksheet/run_bundle_builder.jsx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+              {bundles_html}
+            </div>
+          </div>
+          <div className='run-bundle-container'>
+            <div className='run-bundle-text pop-up-text'>Step 2: Run your command. Your dependencies have been put into the same environment for you. </div>
+            <div id='run-bundle-terminal'>
+              {run_bundle_terminal}
+            </div>
+          </div>
+          <div className='pop-up-text'>Equivalent web terminal command:>
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/run_bundle_builder.jsx]: Remove '>'

In the interface, indent the 'cl' a bit on the next line.

**Comment by percyliang** [comment]
1) Currently, I can only select files inside a bundle as a dependency - but I want to be able to select directories too.
2) The arrow for drilling down into a bundle should be a bit farther away from the name(uuid), and when you hover over it, the mouse cursor should be a hand rather than text.
3) In the black terminal, the text boxes look very jarring.  Can we make the ones for the dependencies have a thinner border, maybe?

**Comment by ChengshuLi** [comment]
@percyliang Have a look :)

**Comment by percyliang** [comment]
1) After run completes, the focus should be set to the worksheet, not the terminal
2) Put 'Step 1:' and 'Step 2:' back
3) The CodaLab> prompt at the bottom shows up on a separate line from 'cl ...'.  They should be on the same line.
4) We need to escape quotes that appear in the command.
5) I can't drill down into bundle directories at all now...

**Comment by ChengshuLi** [comment]
1) Fixed in my new worksheet button PR
2) Done
3) Done
4) Done
5) It seems to be working fine for me. Is your rest-server giving you any error? 

@percyliang

**Comment by percyliang** [comment]
4) if you're using double quotes (") to quote, then don't escape single quotes (')

**Comment by percyliang** [comment]
5) I don't see any errors - sent you a screenshot on slack

**Pull request merged**